baremaps-server-studio/src/main/java/com/baremaps/server/studio/ImportResource.java

```diff
@@ -66,93 +59,73 @@ public Response uploadData(
-    builder.setName(collection.getId().toString());
-    builder.setCRS(schema.getCoordinateReferenceSystem()); // <- Coordinate reference system
-    builder.addAll(schema.getAttributeDescriptors());
-    SimpleFeatureType SCHEMA = builder.buildFeatureType();
+            .title(fileName.substring(0, fileName.lastIndexOf(".") - 1))
+            .extent(
+                new Extent()
+                    .spatial(
+                        new ExtentSpatial()
+                            .bbox(
+                                List.of(
+                                    List.of(
+                                        fc.getBounds().getMinX(),
+                                        fc.getBounds().getMinY(),
+                                        fc.getBounds().getMaxX(),
+                                        fc.getBounds().getMaxY())))))
+            .count(fc.size())
+            .created(new Date())
+            .geometryType(
+                fc.getSchema().getGeometryDescriptor().getType().getBinding().getSimpleName());
 
-    // Setup DataStore
-    URI uri =
-        jdbi.withHandle(
-            haandle -> {
-              String url = haandle.getConnection().getMetaData().getURL();
-              logger.debug(url);
-              return URI.create(url.substring(5));
-            });
-    Map<String, String> query =
-        URLEncodedUtils.parse(uri, StandardCharsets.UTF_8.toString()).stream()
-            .collect(Collectors.toMap(NameValuePair::getName, NameValuePair::getValue));
-    Map<String, Object> params = new HashMap<>();
-    params.put("dbtype", "postgis");
-    params.put("host", uri.getHost());
-    params.put("port", uri.getPort());
-    params.put("schema", query.getOrDefault("currentSchema", "public"));
-    params.put("database", uri.getPath().substring(1));
-    params.put("user", query.getOrDefault("user", "postgres"));
-    params.put("passwd", query.getOrDefault("password", "postgres"));
-    params.forEach((key, value) -> logger.debug("Params: " + key + " -> " + value));
-    DataStore dataStore = DataStoreFinder.getDataStore(params);
-    dataStore.createSchema(SCHEMA);
+    // Load data
+    jdbi.useTransaction(
+        handle -> {
+          // Create collection
+          handle
+              .createUpdate(
+                  "insert into collections (id, collection) values (:id, CAST(:collection AS jsonb))")
+              .bind("id", collection.getId())
+              .bindByType("collection", collection, COLLECTION)
+              .execute();
+          // Create table
+          handle.execute(
+              String.format(
+                  "create table \"%s\" (id serial, tags hstore, geom geometry)",
+                  collection.getId()));
+          // Insert features
+          var features = fc.features();
+          while (features.hasNext()) {
+            SimpleFeature feature = (SimpleFeature) features.next();
+            String tags =
```

Comment:
The PostgresPlugin in jdbi supports hstore with Map<String, String> you can probably avoid string concatenation.

https://jdbi.org/apidocs/org/jdbi/v3/postgres/PostgresPlugin.html

Reply:
indeed, applied in https://github.com/baremaps/baremaps/pull/369/commits/b39a544dd2ef0d50fa425762188933ffec91ca33